bug: Crash when accessing an advancement with abnormal path

### Describe the Bug

```java
// Retrieve the root achievement id without path
final String[] rootSegments = rootAdvancementId.getPath().split("/");
final String realRootAchievementId = rootSegments[rootSegments.length - 2];
```
In `ReloadedCriterionProcess.getTranslationKey()`, the mod tries to get the root advancement id, but if the advancement has an "abnormal" path, it can lead to an `ArrayIndexOutOfBoundsException`.

### Steps to Reproduce

- Install AdvancementInfoReloaded
- Install AdoraBuild: Structures
- In game, open the advancements screen, and click on one of AdoraBuild: Structure's advancement icon
- The game crashes

### Expected Behavior

While this is more of an issue of AdoraBuild: Structures (or any mod with a weird advancements architecture), this should not make the game crash.

### Screenshots

_No response_

### Modloader Type

Fabric

### Modloader Version

0.16.5

### Mod Version

0.6.0

### Log Output

First two lines are for debugging purposes and prints `advancementId.getPath()` and `rootAdvancementId.getPath()`. The first line is when clicking on a "normal" advancement, the second line is when clicking on an "abnormal" advancement, causing the crash.

```shell
[21:44:44] [Render thread/INFO]: [STDOUT]: AdvID: adventure/brush_armadillo    RootAdvId: ADVENTURE/ROOT
[21:44:51] [Render thread/INFO]: [STDOUT]: AdvID: suspiciously_dangerous    RootAdvId: OVERWORLD_STRUCTURES
[21:44:52] [Render thread/ERROR]: Reported exception thrown!
net.minecraft.class_148: Rendering screen
	at net.minecraft.class_757.method_3192(class_757.java:922) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1285) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:882) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:256) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:480) [fabric-loader-0.16.5.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.16.5.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.16.5.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:100) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index -1 out of bounds for length 1
	at codes.atomys.advr.ReloadedCriterionProgress.getTranslationKey(ReloadedCriterionProgress.java:154) ~[advancements_reloaded-0.6.0-fabric-1.21.1.jar:?]
	at codes.atomys.advr.ReloadedCriterionProgress.getHumanCriterionName(ReloadedCriterionProgress.java:115) ~[advancements_reloaded-0.6.0-fabric-1.21.1.jar:?]
	at codes.atomys.advr.screens.AdvancementReloadedScreen.drawAdvancementCriterias(AdvancementReloadedScreen.java:595) ~[advancements_reloaded-0.6.0-fabric-1.21.1.jar:?]
	at codes.atomys.advr.screens.AdvancementReloadedScreen.method_25394(AdvancementReloadedScreen.java:463) ~[advancements_reloaded-0.6.0-fabric-1.21.1.jar:?]
	at net.minecraft.class_437.method_47413(class_437.java:117) ~[client-intermediary.jar:?]
	at net.minecraft.class_757.mixinextras$bridge$method_47413$177(class_757.java) ~[client-intermediary.jar:?]
	at net.minecraft.class_757.wrapOperation$zmi000$fabric-screen-api-v1$onRenderScreen(class_757.java:2086) ~[client-intermediary.jar:?]
	at net.minecraft.class_757.method_3192(class_757.java:913) ~[client-intermediary.jar:?]
	... 9 more
```

### Additional Context

_No response_

--- a/common/src/main/java/codes/atomys/advr/ReloadedCriterionProgress.java
+++ b/common/src/main/java/codes/atomys/advr/ReloadedCriterionProgress.java
@@ -4,6 +4,7 @@ import codes.atomys.advr.config.Configuration;
 import codes.atomys.advr.utils.Utils;
 import com.google.common.collect.Lists;
 import java.util.List;
+import java.util.stream.Collectors;
 import net.minecraft.advancements.Advancement;
 import net.minecraft.advancements.AdvancementNode;
 import net.minecraft.advancements.AdvancementProgress;
@@ -136,29 +137,71 @@ public class ReloadedCriterionProgress {
    * Gets the translation key for the criterion name. This is used for the
    * advancement sidebar. The key is formatted as
    * "advancements.{root-advancement-id}.{advancement-id}.criteria.{criterion-name>}"
+   * <p>
+   * This method will return an empty string if the translation key cannot be
+   * retrieved.
+   * </p>
    *
    * @return the translation key for the criterion name
    */
   public String getTranslationKey() {
-    final ResourceLocation advancementId = this.getResourceLocation();
-    final ResourceLocation rootAdvancementId = this.getAdvancementNode().root().holder().id();
     final String criterionName = this.criterion.getPath();
 
-    // Retrieve the achievement id without path
-    final String[] segments = advancementId.getPath().split("/");
-    final String realAchievementId = segments[segments.length - 1];
+    final List<String> parts = List.of(
+        "advancements",
+        this.getAdvancementCategory(),
+        this.getAdvancementIdentifier(),
+        "criteria",
+        criterionName);
 
-    // Retrieve the root achievement id without path
-    final String[] rootSegments = rootAdvancementId.getPath().split("/");
-    final String realRootAchievementId = rootSegments[rootSegments.length - 2];
+    return parts.stream()
+        .filter(part -> part != null && !part.isEmpty()) // Remove null or empty elements
+        .collect(Collectors.joining("."));
+  }
 
-    final String translationKey = "advancements."
-        + realRootAchievementId
-        + "."
-        + realAchievementId
-        + ".criteria." + criterionName;
+  /**
+   * Retrieves the identifier of the advancement for the current criterion.
+   * This is used to get the translation key for the criterion name in the
+   * sidebar.
+   *
+   * @return the identifier of the advancement
+   */
+  private String getAdvancementIdentifier() {
 
-    return translationKey;
+    final ResourceLocation locationId = this.getResourceLocation();
+    final String path = locationId.getPath();
+
+    final String[] pathSegments = path.split("/");
+    if (pathSegments.length == 0) {
+      Utils.LOGGER.warn("advancement cannot be retrieved from the path: " + path);
+      return "";
+    }
+
+    return pathSegments[pathSegments.length - 1];
+  }
+
+  /**
+   * Retrieves the category of the advancement for the current criterion.
+   * The category are the root advancement of the advancement if there is one.
+   * This is used to get the translation key for the criterion name in the
+   * sidebar.
+   *
+   * @return the category of the advancement
+   */
+  private String getAdvancementCategory() {
+    final ResourceLocation locationId = this.getAdvancementNode().root().holder().id();
+    final String path = locationId.getPath();
+
+    if (!path.contains("/")) {
+      return "";
+    }
+
+    final String[] pathSegments = path.split("/");
+    if (pathSegments.length >= 2) {
+      return pathSegments[pathSegments.length - 2];
+    }
+
+    return "";
   }
 
   private Component retrieveTranslationOnGame() {

--- a/docs/changelogs/0.6.1.md
+++ b/docs/changelogs/0.6.1.md
@@ -4,3 +4,4 @@
 ### 🐛 Fixes
 
 - **Configuration in NeoForge**: Fixed an issue where the configuration in NeoForge was not being saved correctly. [PR #31](https://github.com/42atomys/mc-advancements-reloaded/pull/31)
+- **Auto Translate**: Fixed an issue where the auto-translate feature causing crashes when the achievement don't have a root achievement attached. [PR #32](https://github.com/42atomys/mc-advancements-reloaded/pull/32)

